index.js

```diff
@@ -55,6 +57,7 @@ function ensureAllCommentsPrinted(astComments) {
 }
 
 function formatWithCursor(text, opts, addAlignmentSize) {
+  text = prependFormatIfNecessary(text);
```

Comment:
> do we have any such files which include a BOM?
We probably don't, but Prettier has a feature that strips the BOM when it's present. So let's not break it without reason -- especially in a PR that has a high potential to be accepted also in the official version.